Source/Core/DolphinQt2/Config/ControllersWindow.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QBoxLayout>
+#include <QCheckBox>
+#include <QComboBox>
+#include <QDialog>
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QVBoxLayout>
+
+#include <unordered_map>
+
+#include "Core/HW/SI/SI.h"
+#include "Core/HW/Wiimote.h"
+
+class ControllersWindow final : public QDialog
+{
+  Q_OBJECT
+public:
+  explicit ControllersWindow(QWidget* parent);
+private slots:
+  void OnWiimoteModeChanged(bool passthrough);
+  void OnWiimoteTypeChanged(int state);
+  void OnGCTypeChanged(int state);
+  void SaveSettings();
+  void UnimplementedButton();
+  void OnBluetoothPassthroughSyncPressed();
+  void OnBluetoothPassthroughResetPressed();
+
+private:
+  void CreateGamecubeLayout();
+  void CreateWiimoteLayout();
+  void CreateAdvancedLayout();
+  void CreateMainLayout();
+  void ConnectWidgets();
+  void LoadSettings();
+
+  // Icons
+  QString m_configure_icon;
+  QString m_gamecube_icon;
+  QString m_wii_icon;
+
+  // Main
+  QVBoxLayout* m_main_layout;
+  QHBoxLayout* m_main_hbox;
+
+  // Gamecube
+  QGroupBox* m_gc_box;
+  QLabel* m_gc_label;
+  QFormLayout* m_gc_layout;
+  QComboBox* m_gc_controller_boxes[4];
```

Comment:
These three can be `std::array` instances. It'll also let you get rid of the magic `4` constants in the cpp file in some places